apps/showcase/src/app/modules/rxidb-showcase/components/rxidb-auto-increment/rxidb-auto-increment.component.ts

```diff
@@ -0,0 +1,174 @@
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import {
+  addItem,
+  connectIndexedDb,
+  deleteIndexedDb,
+  deleteItem,
+  entries,
+  getObjectStore,
+  keys,
+  read,
+} from '@this-dot/rxidb';
+import { isTruthy } from '@this-dot/utils';
+import {
+  BehaviorSubject,
+  combineLatest,
+  concatMap,
+  EMPTY,
+  from,
+  map,
+  Observable,
+  Subject,
+  switchMap,
+  takeLast,
+  tap,
+} from 'rxjs';
+import { filter, take, takeUntil } from 'rxjs/operators';
+
+const DATABASE_NAME = 'AUTO_INCREMENT';
+
+@Component({
+  selector: 'this-dot-rxidb-auto-increment',
+  templateUrl: './rxidb-auto-increment.component.html',
+  styleUrls: ['./rxidb-auto-increment.component.scss'],
+  providers: [
+    {
+      provide: 'STORE',
+      useValue: connectIndexedDb(DATABASE_NAME).pipe(
+        getObjectStore('store', { autoIncrement: true })
+      ),
+    },
+  ],
+})
+export class RxidbAutoIncrementComponent implements OnInit, OnDestroy {
+  private destroy$ = new Subject<void>();
+  private isLoadingSubject = new BehaviorSubject(false);
+  private readonly keys$ = this.store$.pipe(keys());
+  private readonly entries$ = this.store$.pipe(entries());
+  readonly isLoading$ = this.isLoadingSubject.asObservable();
+  readonly keyValues$ = combineLatest([this.keys$, this.entries$]).pipe(
+    map(([keys, values]) =>
+      keys.map((key: IDBValidKey, index: number) => ({
+        key,
+        value: values[index],
+      }))
+    )
+  );
+
+  readonly inputControl = new FormControl('');
+
+  constructor(
+    private snackbar: MatSnackBar,
+    @Inject('STORE') private store$: Observable<IDBObjectStore>
+  ) {}
+
+  ngOnInit() {
+    this.isLoading$
+      .pipe(
+        tap((isLoading: boolean) => {
+          if (isLoading) {
+            this.inputControl.disable();
+          } else {
+            this.inputControl.enable();
+          }
+        }),
+        takeUntil(this.destroy$)
+      )
+      .subscribe();
+
+    this.store$
+      .pipe(
+        takeUntil(this.destroy$),
+        tap({
+          complete: () => {
+            this.inputControl.disable();
+          },
+        })
+      )
+      .subscribe();
+  }
+
+  ngOnDestroy() {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  append(evt: any) {
+    evt.preventDefault();
+    this.store$
+      .pipe(
+        addItem(this.inputControl.value),
```

Comment:
Also getting a type error on `addItem` here:
Argument of type '(s$: Observable<IDBObjectStore>) => Observable<IDBObjectStore>' is not assignable to parameter of type 'OperatorFunction<IDBObjectStore, IDBObjectStore>'.

Reply:
I don't get this type error. Neither locally nor in stackblitz. 👀